datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -296,7 +296,9 @@ impl PhysicalExpr for BinaryExpr {
             Operator::Minus => return apply(&lhs, &rhs, sub_wrapping),
             Operator::Multiply if self.fail_on_overflow => return apply(&lhs, &rhs, mul),
             Operator::Multiply => return apply(&lhs, &rhs, mul_wrapping),
-            Operator::Divide => return apply(&lhs, &rhs, div),
+            Operator::Divide => {
+                return apply(&lhs, &rhs, div);
```

Comment:
@andygrove , should we add a new function in arrow(like `div_zero`) to return None. It seems  a bit difficult to rewrite `div` in `binary.rs`.